bioinfokit/visuz.py

```diff
@@ -577,15 +577,18 @@ def geneplot_mhat(df, markeridcol, chr, pv, gwasp, markernames, gfont, gstyle, a
         if markeridcol is not None:
             if markernames is not None and markernames is True:
                 for i in df[markeridcol].unique():
-                    if df.loc[df[markeridcol] == i, pv].iloc[0] <= gwasp:
-                        if gstyle == 1:
-                            plt.text(df.loc[df[markeridcol] == i, 'ind'].iloc[0], df.loc[df[markeridcol] == i, 'tpval'].iloc[0],
-                                    str(i), fontsize=gfont)
-                        elif gstyle == 2:
-                            plt.annotate(i, xy=(df.loc[df[markeridcol] == i, 'ind'].iloc[0], df.loc[df[markeridcol] == i, 'tpval'].iloc[0]),
-                                         xycoords='data', xytext=(5, -15), textcoords='offset points', size=6,
-                                         bbox=dict(boxstyle="round", alpha=0.2),
-                                         arrowprops=dict(arrowstyle="wedge,tail_width=0.5", alpha=0.2, relpos=(0, 0)))
+                    for j in range(len(df[chr].unique())):
```

Comment:
This for-loop enables the annotator iterates all the chromosome, so that it won't only mark the first chromosome.